test/integration/auth.spec.ts

```diff
@@ -168,6 +168,138 @@ describe('admin.auth', () => {
+
+    before(async () => {
+      // Delete all the users that we're about to create (in case they were
+      // left over from a prior run).
+      const uidsToDelete = usersToCreate.map((user) => user.uid);
+      uidsToDelete.push(importUser1.uid);
+      await admin.auth().deleteUsers(uidsToDelete);
+
+      // Create/import users required by these tests
+      await Promise.all(usersToCreate.map((user) => admin.auth().createUser(user)));
+      await admin.auth().importUsers([importUser1]);
+    });
+
+    after(async () => {
+      const uidsToDelete = usersToCreate.map((user) => user.uid);
+      uidsToDelete.push(importUser1.uid);
+      await admin.auth().deleteUsers(uidsToDelete);
+    });
+
+    it('returns users by various identifier types in a single call', async () => {
+      const users = await admin.auth().getUsers([
+          { uid: 'uid1' },
+          { email: 'user2@example.com' },
+          { phoneNumber: '+15555550003' },
+          { providerId: 'google.com', providerUid: 'google_uid4' },
+        ])
+        .then((getUsersResult) => getUsersResult.users)
+        .then(mapUserRecordsToUidEmailPhones);
+
+      expect(users).to.have.deep.members([testUser1, testUser2, testUser3, testUser4]);
+    });
+
+    it('returns found users and ignores non-existing users', async () => {
+      const users = await admin.auth().getUsers([
+          { uid: 'uid1' },
+          { uid: 'uid_that_doesnt_exist' },
+          { uid: 'uid3' },
+        ]);
+      expect(users.notFound).to.have.deep.members([{uid: 'uid_that_doesnt_exist'}]);
+
+      const foundUsers = mapUserRecordsToUidEmailPhones(users.users);
+      expect(foundUsers).to.have.deep.members([testUser1, testUser3]);
+    });
+
+    it('returns nothing when queried for only non-existing users', async () => {
+      const notFoundIds = [{uid: 'non-existing user'}];
+      const users = await admin.auth().getUsers(notFoundIds);
+
+      expect(users.users).to.be.empty;
+      expect(users.notFound).to.deep.equal(notFoundIds);
+    });
+
+    it('de-dups duplicate users', async () => {
+      const users = await admin.auth().getUsers([
+          { uid: 'uid1' },
+          { uid: 'uid1' },
+        ])
+        .then((getUsersResult) => getUsersResult.users)
+        .then(mapUserRecordsToUidEmailPhones);
+
+      expect(users).to.deep.equal([testUser1]);
+    });
+
+    it('returns users with a lastRefreshTime', async () => {
+      const isUTCString = (s: string): boolean => {
+        return new Date(s).toUTCString() === s;
+      };
+
+      const newUserRecord = await admin.auth().createUser({
+        uid: 'lastRefreshTimeUser',
+        email: 'lastRefreshTimeUser@example.com',
+        password: 'p4ssword',
+      });
+
+      try {
+        // New users should not have a lastRefreshTime set.
+        expect(newUserRecord.metadata.lastRefreshTime).to.be.null;
+
+        // Login to cause the lastRefreshTime to be set.
```

Comment:
Suggest:
"// Log in to set the lastRefreshTime."

Reply:
Done.